### PR TITLE
[Agent] Make default config OS agnostic

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

PR changes path of `install` and `download` directories in default config to be OS agnostic by using `path.data`.

## Why is it important?

It simplifies first run experience when running agent by removing the need to change configuration values which are configured to non existent paths.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works

## How to test this PR locally

You need to build  and package agent to contain `data/beats` subdirectory containing filebeat and metricbeat archives (tar.gz on linux and darwin). 

Second step is just running agent and verifying that metricbeat was installed to `data/install` and is successfully started

### CODE REVIEW

File: x-pack/agent/_meta/agent.docker.yml [hunk 1]
```diff
@@ -52,15 +52,15 @@ download:
   # e.g /windows-x86.zip
   sourceURI: "https://artifacts.elastic.co/downloads/beats/"
   # path to the directory containing downloaded packages
-  target_directory: "/home/elastic/downloads"
+  target_directory: "${path.data}/beats"
```
Comment (ruflin) [inline x-pack/agent/_meta/agent.docker.yml]: Perhaps still call it `downloads` and not beats as this will also download endpoint?

Comment (ph) [reply x-pack/agent/_meta/agent.docker.yml]: +1 for download

**Comment by michalpristas** [comment]
`/downloads` sounds good to me as well i changed packaging to inject beats into downloads so it is usable right away. this changes downloader to use `downloads` subdir as a default fallback instead of `beats`

**Comment by michalpristas** [comment]
going to test packaging before merge

**Comment by ph** [comment]
added need backport label.

**Comment by ph** [comment]
backported in https://github.com/elastic/beats/pull/17574 to have a working build

**Pull request merged**